### DISCUSSION TITLE
InvalidArgument in case of more specific argument in child class constructor

### DISCUSSION THREAD

**Discussion by andrew-demb** [vimeo/psalm#7816] [Q&A] [answered]
https://psalm.dev/r/a6deb3aca7

Expected: no issue

Actual: `ERROR: [InvalidArgument](https://psalm.dev/004) - 12:28 - Argument 1 of AbstractConfirmCommand::__construct expects RequestSenderInterface<Iterator>, ConfirmRequestSender provided`

Why psalm is considered this case as an error?
The more specific type is specified in the child constructor just to simplify DI configs (by using autowiring).

**Comment by orklah** [comment]
probably a bug

**Comment by andrew-demb** [comment, marked as answer]
Answered in the issue: https://github.com/vimeo/psalm/issues/7821#issuecomment-1079726499

TLDR: `@template-covariant` fixes an error https://psalm.dev/r/cc94a91509
